tutorials/graphics/saveall.C

```diff
@@ -0,0 +1,36 @@
+/// \file
+/// \ingroup tutorial_graphics
+/// \notebook
+/// This macro creates 100 canvases and store them in different images files using TCanvas::SaveAll() method
+/// Demonstrated how different output format can be used in batch mode.
+///
+/// \macro_code
+///
+/// \author Sergey Linev
+
+void saveall()
+{
+   gROOT->SetBatch(kTRUE); // enforce batch mode to avoid appearance of multiple canvas windows
+
+   std::vector<TPad *> pads;
+
+   for(int n = 0; n < 100; ++n) {
+      auto c = new TCanvas(TString::Format("canvas%d", n), "Canvas with histogram");
+
+      auto h1 = new TH1I(TString::Format("hist%d", n), "Histogram with random data", 100, -5., 5);
```

Comment:
Can we stop showing bare `new` in modern tutorials? In this case the histogram will also be leaked IIUC

Reply:
In my mind it is the only easy way to manage vector of canvases. That will be the alternative? 
`unique_ptr`? But canvases typically remains in memory when macro ends to run - for interactivity.

Reply:
Well, I still think that the leakage of the `TH1I` in this example can be avoided. Since `SetDirectory(nullptr)` is called, I don't see why we can't just declare it on the stack.

As for the leaking of the canvases on purpose, I see the point, but I would still wish we moved away from this pattern at some point. Fingers crossed for ROOT7 I guess.

Reply:
Histograms belong to canvas - via list of primitives. And they will be deleted together with canvases when ROOT session is closed.